### PR TITLE
Do not check for quote style anymore

### PR DESCRIPTION
**Pull request by jyurek**
Everyone has an opinion on quote style.
EVERYONE has an opinion on quote style.
No one's opinion is the same. So we have gnar-style telling us to use double
quotes. And client projects telling us to use single quotes. And other
projects telling us to use double again.

This is very immaterial. They're all strings. The objects are 100%
interchangeable with each other. They both have specification requirements
that do not allow them in certain circumstances (single can't interpolate,
double need escaping more, and so on). There is no one true way. Consistency
doesn't buy us anything here.

Having CI break because of this is really quite frustrating because of how
trivial the issue is. Sure I can run `rubocop --fix` but what if I never had
to?

### CODE REVIEW

**Comment by violet-graves** [comment]
i personally dont care about single vs double strings, but i do like to see only one or the other. is that a possible addition we could make?

**Comment by BlakeWilliams** [comment]
I'm for keeping the rule in for similar reasons to @michael-jacques. I don't care if it's double or single, but I do think it's important to stay consistent with one or the other.

**Comment by kevin-j-m** [comment]
To be clear/fair, none of this particular comment will address the proposed change itself.

> EVERYONE has an opinion on quote style.

The purpose and goal of this repo is to have a defined set of style goals to strive for when appropriate in the given situation, identified as a group. Granted, the *vast* majority of this is rubocop defaults. This, in fairness, is one of those tweaks.

> client projects telling us to use single quotes

If you're working on an existing project that has different standards or a group that adheres to different conventions, then after understanding those conventions, if they don't have a style checker, and would like to use this, but for the items that don't conform to their conventions, you can [override](https://github.com/TheGnarCo/gnar-style#overriding-styles) at the repo level. Otherwise, perhaps gnar-style isn't a fix for the existing group?

> Having CI break because of this

This is a problem that can, and should, be solved without turning a check off. It's never been my (personal) intention to have this style guide adhered to dogmatically, and even less so to force zero variance in service of a computer disallowing one from doing so otherwise. 

Many of us have worked on projects where builds break and work can't be merged due to some style rule that shouldn't apply in a particular scenario, but must be done to satisfy the computer. The intent for this repo is not to perpetuate that. It's intentionally meant to be used as a style guide (which perhaps we can and should be more clear about in the README) to guide the developer and conversation towards a particular ideal, to the extent that static checkers can define an ideal and identify variance from it.

That being said, that variance  can and should be allowed for. There are situations where it's warranted. And it's not my hope that this repo disallows for that. It may spark a conversation, and perhaps a spirited one, and there may be all sorts of reasons a particular suggestion from gnar-style should be ignored in a particular scenario. 

> Sure I can run `rubocop --fix` but what if I never had to?

Taking this through to an exaggerated conclusion, it could be interpreted that the suggestion be that we not have a style guide at all (Note: I know that's not what you're implying, I hope this isn't taken in an inflammatory manner. If so, I apologize. Sincerely.). A particular project can also choose to not use or adhere to this guide, holistically or in a particular instance, but the point is to provide a basis which we can build upon of shared conventions we (largely) agree with, to the extent they make sense in the particular instance.

File: rubocop/rubocop.yml [hunk 1]
```diff
@@ -1,8 +1,3 @@
-Style/StringLiterals:
-  EnforcedStyle: double_quotes
-  Exclude:
-    - 'Gemfile'
-
```
Comment (kevin-j-m) [inline rubocop/rubocop.yml:5]: I don't believe this change has the intended effect you're proposing. This will revert back to the rubocop default, which is single-quotes.

**Review by kevin-j-m — COMMENTED** [review]
TL;DR: I'm not in favor of this change, but I am but one voice to be heard from and considered.

I *personally* am in favor of using double-quotes as the standard (which, per a previous comment, can and should be deviated from in particular instances where it makes sense to). That is because there's, as far as I know, no longer any performance hits from using double-quotes (I think there were in ruby versions from years gone-by), and *most* of the time, what I want to do can be done in double-quotes, and more things that I can/need/want to do (interpolation) require double-quotes than single-quotes.

In service of a convention to serve as a baseline, as with other style suggestions, we've stuck with double-quotes to this time. It seems there's appetite to have consistency of some kind (again, unless there's a particular reason to stray from it, like a string that has a bunch of (or any) `"` characters in it, for example). That's been the primary driver in support of this convention, to give a historical context.

That doesn't mean it's set in stone. As with any code/repo, this is a living document and we welcome suggestions, improvements (definitely more subjective in this repo), and differences so that we can have those conversations and hopefully build consensus around it.

I appreciate this contribution, and I hope we get a lot of good conversation on it so we can come to some agreement on a path forward. This merging wouldn't be the first style rule in this guide that I personally disagree with. Should it merge, I doubt it'll be the last.

File: rubocop/rubocop.yml (hunk 1 above)
Comment (jyurek) [reply rubocop/rubocop.yml:5]: I knew something like that would happen. >_<

**Comment by jyurek** [comment]
Part of this was blowing off steam from a really bad commute and then having my build rejected for rubocop errors. But part of this is that I do believe that while having a consistent style is important, some things are ultimately so trivial that we don't even need to automate them. A guide against ternaries instead of if/else? Cool (even if rubocop can fix it automatically). String delimiters? Why does this even register?

I realize that the meat of this problem is that the client's requirement and our requirement don't align. But I feel that taking this out (correctly anyway) is us planting a flag saying that this isn't important, rather than passively accepting either... if that differentiation makes sense. Even if we can't change other people's opinions (and we're professionals, we can deal), we can say that the actual consideration here isn't important.

**Comment by kevin-j-m** [comment]
@BlakeWilliams @michael-jacques you've both stated that you're in favor of consistency on this, in one way or another. Could you expand on that for the group?

I'm not intentionally pitting people against each other, but @jyurek provided more context with his justification, so I want to make sure whatever solution we come to addresses everyone's needs, to the extent possible.

**Comment by BlakeWilliams** [comment]
I could be wrong, but after reading through all of this there seems to be more of an issue with Rubocop failing builds than with single vs double quotes.

If that's the case, I totally agree. We shouldn't have builds fail because of Rubocop, but we should be informed when we violate the style guide (rubocop).

> This is very immaterial. They're all strings. The objects are 100%
interchangeable with each other. They both have specification requirements
that do not allow them in certain circumstances (single can't interpolate,
double need escaping more, and so on).

This is true, but this is how the computer interprets it, not people.

>  There is no one true way. Consistency
doesn't buy us anything here.

I think consistency is a good thing. With consistency we have something to expect which means we can free ourselves of asking "should this be double quote or single?" since it has already been decided for us. I think if we remove this rule, we'll think we'll free ourselves from that question but a desire for consistency will bring us back to that question repeatedly (as it has already done to many a team).

**Comment by kevin-j-m** [comment]
@jyurek how would you feel about closing this? Do we still need further discussion to push this towards a resolution?

**Comment by jyurek** [comment]
Yeah I think we can close it.
